wfl/Database/MongoDBManager.cs

```diff
@@ -19,62 +20,66 @@ public MongoDBManager()
+                }
+                else
+                    collectionName += c;
+            }
+            collectionName += 's';
+            Console.WriteLine(collectionName);
+            return collectionName;
 
-        public async Task<Restaurant> UpdateRestaurantAsync(string id, RestaurantDto r)
-        {
-            IMongoCollection<Restaurant> collection = db.GetCollection<Restaurant>("restaurants");
-            var filter = Builders<Restaurant>.Filter.Eq("Id", id);
-            Restaurant restaurant = new Restaurant
-            {
-                Name = r.Name,
-                MenuLink = r.MenuLink
-            };
-            await collection.ReplaceOneAsync(filter, restaurant);
-            IAsyncCursor<Restaurant> task = await collection.FindAsync(filter);
-            return await task.FirstAsync();
         }
 
-        public async Task<IEnumerable<Restaurant> > GetAllRestaurantsAsync()
+        public async Task<T> InsertEntityAsync<T>(T entity) where T: IEntity, new()
         {
-            IMongoCollection<Restaurant> collection = db.GetCollection<Restaurant>("restaurants");
-            IAsyncCursor<Restaurant> task = await collection.FindAsync(r => true, null);
-            return task.ToEnumerable<Restaurant>();
+            IMongoCollection<T> collection = db.GetCollection<T>(GenerateCollectionName<T>());
+            await collection.InsertOneAsync(entity);
+            return entity;
         }
 
-        public async Task<Restaurant> GetRestaurantWithIdAsync(string Id)
+        public async Task<IEnumerable<T>> GetAllEntitiesAsync<T>() where T: IEntity, new()
         {
-            IMongoCollection<Restaurant> collection = db.GetCollection<Restaurant>("restaurants");
-            IAsyncCursor<Restaurant> task = await collection.FindAsync(r => r.Id == Id, null);
-            return await task.FirstOrDefaultAsync();
+            IMongoCollection<T> collection = db.GetCollection<T>(GenerateCollectionName<T>());
+            IAsyncCursor<T> task = await collection.FindAsync(e => true, null);
+            return task.ToEnumerable();
         }
 
-        public async Task<IEnumerable<FoodOrder>> GetAllFoodOrdersAsync()
+        public async Task<T> GetEntityWithIdAsync<T>(string id) where T: IEntity, new()
         {
-            IMongoCollection<FoodOrder> collection = db.GetCollection<FoodOrder>("food-orders");
-            IAsyncCursor<FoodOrder> task = await collection.FindAsync(order => true, null);
-            return task.ToEnumerable<FoodOrder>();
+            IMongoCollection<T> collection = db.GetCollection<T>(GenerateCollectionName<T>());
+            IAsyncCursor<T> task = await collection.FindAsync(e => e.Id == id, null);
+            return await task.FirstOrDefaultAsync();
         }
 
-        public async Task<FoodOrder> InsertFoodOrderAsync(FoodOrderDto o)
+        public async Task<T> UpdateEntityAsync<T>(string id, T entity) where T : IEntity, new()
```

Comment:
I assume another step in the repository pattern would be to remove the `string id` parameter since we can now do `entity.Id` to get the id?? is that the thought?